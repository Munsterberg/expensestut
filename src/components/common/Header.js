import React from 'react';
import { NavLink } from 'react-router-dom';

function Header() {
  return (
    <header>
      <ul>
        <NavLink to="/" activeClassName="active-link"><li>Home</li></NavLink>
        <NavLink to="/add" activeClassName="active-link"><li>Add Expense</li></NavLink>
      </ul>
    </header>
  );
}

export default Header;
