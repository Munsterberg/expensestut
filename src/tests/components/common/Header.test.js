import React from 'react';
import { shallow } from 'enzyme';

import Header from '../../../components/common/Header';

let wrapper = null;

beforeEach(() => {
  wrapper = shallow(<Header />);
});

test('should render <Header /> correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should be type type of header', () => {
  expect(wrapper.type()).toBe('header');
});

test('should render ul', () => {
  expect(wrapper.find('ul')).toHaveLength(1);
});