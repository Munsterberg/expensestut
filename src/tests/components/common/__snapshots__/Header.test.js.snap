// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <Header /> correctly 1`] = `
<header>
  <ul>
    <NavLink
      activeClassName="active-link"
      ariaCurrent="true"
      to="/"
    >
      <li>
        Home
      </li>
    </NavLink>
    <NavLink
      activeClassName="active-link"
      ariaCurrent="true"
      to="/add"
    >
      <li>
        Add Expense
      </li>
    </NavLink>
  </ul>
</header>
`;
